src/renderer/stores/Notebook.js

```diff
@@ -10,29 +10,36 @@ var Notebook = Backbone.Collection.extend({
   initialize: function() {
     this.dispatchToken = JotzDispatcher.register(this.dispatchCallback.bind(this));
     // listen for ipc completion event on client
-    ipc.on('save-note-reply', function(arg) { console.log(arg + ' caught reply on renderer'); });
+    // TODO: integrate with Clark
+    // ipc.on('save-note-reply', function(arg) { console.log(arg + ' caught reply on renderer'); });
   },
 
   dispatchCallback: function(payload) {
-    switch(payload.action.type) {
-      case 'make-note':
-        this.saveNote(new Note(payload.action.content));
+    switch(payload.actionType) {
+      case 'create-note':
+        this.add({content: payload.content})
```

Comment:
should the note be added to the collection before written to file? if there is an I/O error on the file system side, then we would have to pull it out of the collection. I suggest only adding the note to the collection after the ipc reply has come back with a result status (i.e., listening for the reply in initialize (as is commented out). Pushback for optimistically adding to collection? Thoughts? @omarCA @d4ncer @minocys